bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -263,23 +271,37 @@ def __del__(self):
     def _gen_run_method(runnable_self: TensorflowRunnable, method_name: str):
         raw_method = getattr(runnable_self.model, method_name)
         method_partial_kwargs = partial_kwargs.get(method_name)
-        if method_partial_kwargs:
-            raw_method = functools.partial(raw_method, **method_partial_kwargs)
-
-        def _mapping(item: "TFArgType") -> "tf_ext.TensorLike":
-            if not LazyType["tf_ext.TensorLike"]("tensorflow.Tensor").isinstance(item):
-                return t.cast("tf_ext.TensorLike", tf.convert_to_tensor(item))
-            else:
-                return item
 
         def _run_method(
             _runnable_self: TensorflowRunnable,
             *args: "TFArgType",
             **kwargs: "TFArgType",
         ) -> "ext.NpNDArray":
-            params = Params["TFArgType"](*args, **kwargs)
-            params = params.map(_mapping)
-            res = raw_method(*params.args, **params.kwargs)
+            try:
+                if method_partial_kwargs is None:
+                    res = raw_method(*args, **kwargs)
+                else:
+                    res = raw_method(*args, **dict(method_partial_kwargs, **kwargs))
+            except ValueError:  # try to cast to model signatures
+                sigs = get_input_signatures(raw_method)
+                if not sigs:
+                    raise
+                arg_specs, kwarg_specs = sigs[0]
+                trans_args: t.Tuple[t.Any, ...] = tuple(
+                    cast_tensor_by_spec(arg, spec) for arg, spec in zip(args, arg_specs)  # type: ignore[arg-type]
+                )
+
+                trans_kwargs = {
+                    k: cast_tensor_by_spec(arg, kwarg_specs[k])
+                    for k, arg in kwargs.items()
+                }
+                if method_partial_kwargs is None:
+                    res = raw_method(*trans_args, **trans_kwargs)
+                else:
+                    res = raw_method(
+                        *trans_args,
+                        **dict(method_partial_kwargs, **trans_kwargs),
```

Comment:
```suggestion
                        **trans_kwargs,
                        **method_partial_kwargs,
```

Reply:
same as above

Reply:
at this stage should we also try to cast `method_partial_kwargs`?